File: app/src/main/java/com/teambrella/android/ui/claim/ClaimDetailsFragment.java
package com.teambrella.android.ui.claim;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.google.gson.JsonObject;
import com.teambrella.android.R;
import com.teambrella.android.api.TeambrellaModel;
import com.teambrella.android.api.model.json.JsonWrapper;
import com.teambrella.android.ui.base.ADataFragment;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Locale;

import io.reactivex.Notification;

/**
 * Claims Details Fragment
 */
public class ClaimDetailsFragment extends ADataFragment<IClaimActivity> {


    private TextView mClaimAmount;
    private TextView mExpenses;
    private TextView mDeductible;
    private TextView mCoverage;
    private TextView mInsidentDate;
    private static SimpleDateFormat mDateFormat = new SimpleDateFormat("d LLLL", Locale.ENGLISH);
    private static SimpleDateFormat mSDF = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.US);


    public static ClaimDetailsFragment getInstance(String[] dataTags) {
        return ADataFragment.getInstance(dataTags, ClaimDetailsFragment.class);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_claim_details, container, false);
        mClaimAmount = (TextView) view.findViewById(R.id.claim_amount);
        mExpenses = (TextView) view.findViewById(R.id.estimated_expenses);
        mDeductible = (TextView) view.findViewById(R.id.deductible);
        mCoverage = (TextView) view.findViewById(R.id.coverage);
        mInsidentDate = (TextView) view.findViewById(R.id.incident_date);
        return view;
    }

    @Override
    protected void onDataUpdated(Notification<JsonObject> notification) {
        if (notification.isOnNext()) {
            JsonWrapper response = new JsonWrapper(notification.getValue());
            JsonWrapper data = response.getObject(TeambrellaModel.ATTR_DATA);
            JsonWrapper basic = data.getObject(TeambrellaModel.ATTR_DATA_ONE_BASIC);
            if (basic != null) {
                mClaimAmount.setText(getString(R.string.amount_format_string, Math.round(basic.getDouble(TeambrellaModel.ATTR_DATA_CLAIM_AMOUNT, 0f))));
                mExpenses.setText(getString(R.string.amount_format_string, Math.round(basic.getDouble(TeambrellaModel.ATTR_DATA_ESTIMATED_EXPENSES, 0f))));
                mDeductible.setText(getString(R.string.amount_format_string, Math.round(basic.getDouble(TeambrellaModel.ATTR_DATA_DEDUCTIBLE, 0f))));
                mCoverage.setText(getString(R.string.percentage_format_string, Math.round(basic.getDouble(TeambrellaModel.ATTR_DATA_COVERAGE, 0f) * 100)));
                try {
                    mDataHost.setSubtitle(mDateFormat.format(mSDF.parse(basic.getString(TeambrellaModel.ATTR_DATA_INCIDENT_DATE))));
                } catch (ParseException e) {
                    e.printStackTrace();
                }
            }


        }
    }
}

File: app/src/main/java/com/teambrella/android/ui/claim/ClaimDetailsFragment.kt
package com.teambrella.android.ui.claim

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.google.gson.JsonObject
import com.teambrella.android.R
import com.teambrella.android.api.*
import com.teambrella.android.ui.base.ADataFragment
import com.teambrella.android.util.AmountCurrencyUtil
import com.teambrella.android.util.TeambrellaDateUtils
import io.reactivex.Notification
import java.text.DecimalFormat

/**
 * Claims Details Fragment
 */
class ClaimDetailsFragment : ADataFragment<IClaimActivity>() {

    private val claimAmountView: TextView? by ViewHolder(R.id.claim_amount)
    private val expensesView: TextView? by ViewHolder(R.id.estimated_expenses)
    private val deductibleView: TextView? by ViewHolder(R.id.deductible)
    private val coverageView: TextView? by ViewHolder(R.id.coverage)
    private val incidentDateView: TextView? by ViewHolder(R.id.incident_date)
    private val decimalFormat = DecimalFormat.getInstance()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
            inflater.inflate(R.layout.fragment_claim_details, container, false)


    override fun onDataUpdated(notification: Notification<JsonObject>) {
        notification.takeIf { it.isOnNext }?.value?.data?.let { _data ->
            val basic = _data.basic
            val sign = _data.team?.currency?.let { AmountCurrencyUtil.getCurrencySign(it) }

            if (basic != null && sign != null) {
                claimAmountView?.text = getString(R.string.amount_format_string, sign
                        , decimalFormat.format(Math.round(basic.claimAmount ?: 0f)))
                expensesView?.text = getString(R.string.amount_format_string, sign
                        , decimalFormat.format(Math.round(basic.estimatedExpenses ?: 0.0)))
                deductibleView?.text = getString(R.string.amount_format_string, sign
                        , decimalFormat.format(Math.round(basic.deductible ?: 0.0)))
                coverageView?.text = getString(R.string.percentage_format_string
                        , Math.round((basic.coverage ?: 0f) * 100))

                val date = TeambrellaDateUtils.getDatePresentation(context, TeambrellaDateUtils.TEAMBRELLA_UI_DATE, basic.incidentDate)
                dataHost.setSubtitle(date)
                incidentDateView?.text = date

            }
        }
    }
}
